pkg/nsx/services/inventory/inventory.go

```diff
@@ -0,0 +1,460 @@
+		} else if pre != nil && !reflect.DeepEqual(pre.(containerinventory.ContainerApplicationInstance).OriginProperties, cur.(containerinventory.ContainerApplicationInstance).OriginProperties) {
+			if s.isIPChanged(pre.(containerinventory.ContainerApplicationInstance), cur.(containerinventory.ContainerApplicationInstance)) {
+				updateProperties["origin_properties"] = cur.(containerinventory.ContainerApplicationInstance).OriginProperties
+			}
+		}
+	}
+	if pre == nil {
+		s.requestBuffer = append(s.requestBuffer, containerinventory.ContainerInventoryObject{ContainerObject: updateProperties, ObjectUpdateType: operation_create})
+		return operation_create, updateProperties
+	} else if len(updateProperties) > 2 {
+		s.requestBuffer = append(s.requestBuffer, containerinventory.ContainerInventoryObject{ContainerObject: updateProperties, ObjectUpdateType: operation_update})
+		return operation_update, updateProperties
+	} else {
+		return operation_none, nil
+	}
+
+}
+func (s *InventoryService) isIPChanged(pre containerinventory.ContainerApplicationInstance, cur containerinventory.ContainerApplicationInstance) bool {
+	preIPs := ""
+	curIPs := ""
+	for _, prop := range pre.OriginProperties {
+		if prop.Key == "ip" {
+			preIPs = prop.Value
+			log.Info("Pod previously has IP ", "pod", pre.DisplayName, "ips", preIPs)
+		}
+	}
+	for _, prop := range cur.OriginProperties {
+		if prop.Key == "ip" {
+			curIPs = prop.Value
+			log.Info("Pod now has IP", "pod", cur.DisplayName, "ips", curIPs)
+		}
+	}
+	preIPArr := strings.Split(preIPs, ",")
+	curIPArr := strings.Split(curIPs, ",")
+	if len(preIPArr) != len(curIPArr) {
+		return true
+	} else if len(curIPArr) == 1 && preIPArr[0] != curIPArr[0] {
+		return true
+	} else if len(curIPArr) == 2 {
+		if !((preIPArr[0] == curIPArr[0] && preIPArr[1] == curIPArr[1]) || (preIPArr[0] == curIPArr[1] && preIPArr[1] == curIPArr[0])) {
+			return true
+		}
+	}
+	return false
+}
+
+type empty struct{}
+type KeySet map[InventoryKey]empty
+
+func (s KeySet) Has(item InventoryKey) bool {
+	_, exists := s[item]
+	return exists
+}
+
+func (s KeySet) Insert(item InventoryKey) {
+	s[item] = empty{}
+}
+
+func (s KeySet) Delete(item InventoryKey) {
+	delete(s, item)
+}
+
+func (s *InventoryService) SyncInventoryObject(bufferedKeys KeySet) (KeySet, error) {
+	retryKeys := KeySet{}
+	startTime := time.Now()
+	defer func() {
+		log.Info("Finished syncing inventory object since ", "time", time.Since(startTime))
+	}()
+	for key := range bufferedKeys {
+		log.V(1).Info("Syncing inventory object", "object key", key)
+		externalId := key.ExternalId
+		namespace, name, err := cache.SplitMetaNamespaceKey(key.Key)
+		if err != nil {
+			log.Error(err, "Failed to split meta namespace key", "key", key)
+			continue
+		}
+		switch key.InventoryType {
+
+		case ContainerApplicationInstance:
+			pod := &corev1.Pod{}
+			err := s.Client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, pod)
```

Comment:
So you keep a inventoryObjectQueue which contains only keys, any by client.get to check if we delete or add.
Why not construct a key and operation(delete/update/create) object into queue at handlePod, then we don't need to Get here again.

Reply:
construct a key and operation doesn't make the logic simple.